pkg/object/hdfs_kerberos.go

```diff
@@ -35,30 +33,26 @@ func getKerberosClient() (*krb.Client, error) {
 	// Try to authenticate with keytab file first.
 	keytabPath := os.Getenv("KRB5KEYTAB")
 	keytabBase64 := os.Getenv("KRB5KEYTAB_BASE64")
+	principal := os.Getenv("KRB5PRINCIPAL")
+
+	var kt *keytab.Keytab
 	if keytabBase64 != "" {
 		decodedKeytab, err := base64.StdEncoding.DecodeString(keytabBase64)
 		if err != nil {
 			return nil, fmt.Errorf("error decoding Base64 encoded data %s", err)
 		}
-		decodedKeytabPath := filepath.Join(os.TempDir(), "decodedKeytab")
-		decodedKeytabFile, err := os.Create(decodedKeytabPath)
-		if err != nil {
-			return nil, fmt.Errorf("failed to create %s", decodedKeytabPath)
-		}
-		defer decodedKeytabFile.Close()
-		// keytab file format uses network byte order
-		err = binary.Write(decodedKeytabFile, binary.BigEndian, decodedKeytab)
+		kt = new(keytab.Keytab)
+		err = kt.Unmarshal(decodedKeytab)
 		if err != nil {
-			return nil, fmt.Errorf("failed to write %s", decodedKeytabPath)
+			return nil, err
 		}
-		keytabPath = decodedKeytabPath
-	}
-	principal := os.Getenv("KRB5PRINCIPAL")
-	if keytabPath != "" && principal != "" {
-		kt, err := keytab.Load(keytabPath)
+	} else if keytabPath != "" {
+		kt, err = keytab.Load(keytabPath)
 		if err != nil {
 			return nil, err
 		}
+	}
+	if kt != nil && principal != "" {
```

Comment:
when KRB5PRINCIPAL is not set, we will ignore  KRB5KEYTAB or KRB5KEYTAB_BASE64, is that expected?